File: burstLibrary/src/main/java/com/harry1453/burst/explorer/entity/Account.java
package com.harry1453.burst.explorer.entity;

public class Account {
    public final BurstAddress address;
    public final String publicKey;
    public final String name;
    public final String description;
    public final BurstValue balance;
    public final BurstValue forgedBalance;
    // todo issued assets
    // todo don't display reward recipient if it is set to the same address
    public final BurstAddress rewardRecipient;
    public final String rewardRecipientName;

    public Account(BurstAddress address, String publicKey, String name, String description, BurstValue balance, BurstValue forgedBalance, BurstAddress rewardRecipientAddress, String rewardRecipientName) {
        this.address = address;
        this.publicKey = publicKey;
        this.balance = balance;
        this.name = name;
        this.description = description;
        this.forgedBalance = forgedBalance;
        this.rewardRecipient = rewardRecipientAddress;
        this.rewardRecipientName = rewardRecipientName;
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/entity/Account.kt
package com.harry1453.burst.explorer.entity

data class Account (
        @JvmField val address: BurstAddress,
        @JvmField val publicKey: String,
        @JvmField val name: String,
        @JvmField val description: String,
        @JvmField val balance: BurstValue,
        @JvmField val forgedBalance: BurstValue,
        // TODO issued assets
        @JvmField val rewardRecipient: BurstAddress,
        @JvmField val rewardRecipientName: String
)